### PR TITLE
Refactor the BasePipeline, move out all Project related logic #1351

### PR DESCRIPTION
**Pull request by tdruez**
Refactor the BasePipeline to remove the scanpipe.Project logic and make it reusable without Project context.

### CODE REVIEW

**Comment by tdruez** [comment]
@keshav-space could you have a look before the merge?

File: aboutcode/pipeline/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# `aboutcode.pipeline`
+
+Define and run pipelines.
+
+### Install
+
+```bash
+pip install aboutcode_pipeline
```
Comment (keshav-space) [inline aboutcode/pipeline/README.md:8]: As per PEP8 https://peps.python.org/pep-0008/#package-and-module-names `aboutcode-pipeline` would be a much more Pythonic package name.
```suggestion
pip install aboutcode-pipeline
```

**Review by keshav-space — APPROVED** [review]
Thanks @tdruez, Looking Good!

File: aboutcode/pipeline/README.md (hunk 1 above)
Comment (tdruez) [reply aboutcode/pipeline/README.md:8]: I initially went with aboutcode_pipeline in the README as the filenames of the flot build were `aboutcode_pipeline-0.1.whl`, even though the [project] name was set to "aboutcode.pipeline".
@pombredanne Any input on this?

**Pull request merged**